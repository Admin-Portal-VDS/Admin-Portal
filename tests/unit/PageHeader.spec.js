import { mount } from '@vue/test-utils';
import PageHeader from '@/components/common/PageHeader.vue';

describe('PageHeader.vue', () => {
  it('renders the heading with the correct text', () => {
    const heading = 'Hello World';
    const wrapper = mount(PageHeader, {
      props: {
        heading,
      },
    });
    const title = wrapper.find('[data-id="heading"]');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe(heading);
  });
  it('accepts props and passes them correctly', () => {
    const heading = 'Test Button';
    const wrapper = mount(PageHeader, {
      propsData: { heading },
    });
    expect(wrapper.props('heading')).toBe(heading);
  });

  it('matches snapshot', () => {
    const wrapper = mount(PageHeader, {
      propsData: {
        heading: 'Snapshot Test', // Correct prop name
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
