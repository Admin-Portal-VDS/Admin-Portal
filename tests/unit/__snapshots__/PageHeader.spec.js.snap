// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue matches snapshot 1`] = `<div data-id="heading" class="heading">Snapshot Test</div>`;
